tests/test_csc.py

```diff
@@ -160,6 +160,107 @@ def test_standard_state_transitions(self):
         # with pytest.raises(AttributeError):
         #     m.previous_state = "ENABLED"
 
+    def test_every_fault_transition(self):
+        """Test standard CSC Fault transitions.
+         Returns
+        -------
+        The initial state is STANDBY.
+        The standard commands and associated state transitions are:
+         * enterControl: OFFLINE to STANDBY
+        * start: STANDBY to DISABLED
+        * enable: DISABLED to ENABLED
+         * disable: ENABLED to DISABLED
+        * standby: DISABLED to STANDBY
+        * exitControl: STANDBY, FAULT to OFFLINE (quit)
+        """
+        
+        async def doit():
+            harness = Harness()
+            self.assertEqual(harness.csc.summary_state, salobj.State.OFFLINE)
+
+            # send enterControl; new state is STANDBY
+            cmd_attr = getattr(harness.remote, f"cmd_enterControl")
+            state_coro = harness.remote.evt_summaryState.next(timeout=1.)
+            id_ack = await cmd_attr.start(cmd_attr.DataType(), timeout=1.)
+            state = await state_coro
+            self.assertEqual(id_ack.ack.ack, harness.remote.salinfo.lib.SAL__CMD_COMPLETE)
+            self.assertEqual(id_ack.ack.error, 0)
+            self.assertEqual(harness.csc.summary_state, salobj.State.STANDBY)
+            self.assertEqual(state.summaryState, salobj.State.STANDBY)
+            # set to FAULT; new state is FAULT
+            harness.csc._summary_state = salobj.State.FAULT
```

Comment:
Transitioning to a fault state and reporting the summary state can be achieved in a single line by using the `summary_state` method of the `BaseCsc` class. Any reason for doing it this way?